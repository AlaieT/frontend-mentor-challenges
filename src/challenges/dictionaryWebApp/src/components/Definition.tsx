import React from "react";
import { NavLink } from "react-router-dom";

import classNames from "../utils/classNames";

import type { DefinitionProps } from "../types";

import styles from "../styles/components/definition.module.scss";

const Definition = ({
  mode,
  partOfSpeech,
  definitions,
  synonyms,
  antonyms
}: DefinitionProps) => (
  <section
    className={classNames({
      [styles.definition]: true,
      [styles.dark]: mode === "dark"
    })}
  >
    <h2>{partOfSpeech}</h2>
    <div className={styles.meanings}>
      <h3>Meaning</h3>
      <ul>
        {definitions.map(({ definition, example }) => (
          <li key={definition}>
            <p className={styles.meaning}>{definition}</p>
            {example && (
              <p className={styles.example}>
                &quot;
                {example}
                &quot;
              </p>
            )}
          </li>
        ))}
      </ul>
    </div>
    {!!synonyms.length && (
      <div className={styles["sym-ant"]}>
        <h3>Synonyms</h3>
        <ul>
          {synonyms.map((synonym) => (
            <li key={synonym}>
              <NavLink to={`/${synonym}`}>{synonym}</NavLink>
            </li>
          ))}
        </ul>
      </div>
    )}
    {!!antonyms.length && (
      <div className={styles["sym-ant"]}>
        <h3>Antonyms</h3>
        <ul>
          {antonyms.map((antonym) => (
            <li key={antonym}>
              <NavLink to={`/${antonym}`}>{antonym}</NavLink>
            </li>
          ))}
        </ul>
      </div>
    )}
  </section>
);

export default Definition;
