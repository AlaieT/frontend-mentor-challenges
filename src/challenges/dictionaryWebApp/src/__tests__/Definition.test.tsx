import React from "react";
import { render, cleanup } from "@testing-library/react";

import Definition from "../components/Definition";

afterEach(cleanup);

describe("Definition", () => {
  const props = {
    partOfSpeech: "noun",
    definitions: [
      {
        definition:
          "An answer or reply, or something in the nature of an answer or reply."
      },
      {
        definition:
          "The act of responding or replying; reply: as, to speak in response to a question."
      }
    ],
    synonyms: ["synonym1", "synonym2"],
    antonyms: ["antonym1", "antonym2"]
  };

  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(<Definition mode="light" {...props} />).asFragment()
      ).toMatchSnapshot();
    });
  });
});
