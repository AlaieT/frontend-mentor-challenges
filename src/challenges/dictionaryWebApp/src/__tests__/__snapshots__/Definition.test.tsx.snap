// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition correct render should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="definition"
  >
    <h2>
      noun
    </h2>
    <div
      class="meanings"
    >
      <h3>
        Meaning
      </h3>
      <ul>
        <li>
          <p
            class="meaning"
          >
            An answer or reply, or something in the nature of an answer or reply.
          </p>
        </li>
        <li>
          <p
            class="meaning"
          >
            The act of responding or replying; reply: as, to speak in response to a question.
          </p>
        </li>
      </ul>
    </div>
    <div
      class="sym-ant"
    >
      <h3>
        Synonyms
      </h3>
      <ul>
        <li>
          synonym1
        </li>
        <li>
          synonym2
        </li>
      </ul>
    </div>
    <div
      class="sym-ant"
    >
      <h3>
        Antonyms
      </h3>
      <ul>
        <li>
          antonym1
        </li>
        <li>
          antonym2
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
